### PR TITLE
Add datadog chart to stable.

### PR DESCRIPTION
**Pull request by gtaylor**
Adds a [DataDog](https://www.datadoghq.com/) chart. Fairly popular hosted monitoring service that works relatively well on Kubernetes.

### CODE REVIEW

**Comment by lachie83** [comment]
Very cool. Will review

**Comment by lachie83** [comment]
Without having an API it makes it really hard to test. Is there an easy way to do that?

**Comment by gtaylor** [comment]
Sign up for an account, pass in your token, check in their web UI.

File: stable/datadog/templates/NOTES.txt
Comment (lachie83) [inline stable/datadog/templates/NOTES.txt]: Can you callout the signup URL?

File: stable/datadog/templates/daemonset.yaml
Comment (lachie83) [inline stable/datadog/templates/daemonset.yaml:3]: I think it would make sense to enable the Kubernetes integration by default on this Helm chart. Probably would make sense to load that in via configMap

**Review by lachie83 — CHANGES_REQUESTED** [review]
Add signup URL and enable Kubernetes integration by default

File: stable/datadog/templates/daemonset.yaml
Comment (irabinovitch) [reply stable/datadog/templates/daemonset.yaml:3]: Would it make sense to enable Service Discovery with the Docker backend as well? And possible add a dependency on kube-state-metrics?

Comment (gtaylor) [reply stable/datadog/templates/daemonset.yaml:3]: @irabinovitch I'm not sure that I understand. This DaemonSet's Agent sends a whole bunch of metrics and events for Kubernetes and the underlying containerizers. Enabling Kubernetes reporting pulls in a handful of other Docker things. No further knob twiddling should be needed.

File: stable/datadog/README.md
Comment (lachie83) [inline stable/datadog/README.md:36]: I'm not sure if this meets the documentation requirements given --https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#documentation-requirements

Specifically 

> Customization: explaining options in values.yaml and their defaults

File: stable/datadog/templates/daemonset.yaml
Comment (irabinovitch) [reply stable/datadog/templates/daemonset.yaml:3]: @gtaylor  If you set the environment variable SD_BACKEND=docker it will enable our [service discovery functionality](https://www.datadoghq.com/blog/service-discovery-monitoring-docker-containers-move/) in the agent such that it automatically monitors new containers for the correct applications as they are started/stopped/rescheduled.

My comment on [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) relates to the kubernetes_state integration which was added in the 5.10 release. If a kube-state-metrics pod is online and seen by service discovery, we'll pull in all the metrics that kube-state-metrics exposes.

**Comment by gtaylor** [comment]
Ahh, I had no idea. Thanks for the explanation. I'll get this added!

File: stable/datadog/templates/daemonset.yaml
Comment (lachie83) [reply stable/datadog/templates/daemonset.yaml:3]: FYI @gtaylor - you add a flag to install the kube-state-metrics. Here's the deployment from the coreos prometheus operator.

https://coreos.com/operators/prometheus/latest/exporters.yaml

**Comment by lachie83** [comment]
This just keeps getting better!

File: stable/datadog/templates/NOTES.txt
Comment (irabinovitch) [reply stable/datadog/templates/NOTES.txt]: @gtaylor Sent a PR to your repo adding the signup instructions.

File: stable/datadog/templates/daemonset.yaml
Comment (irabinovitch) [reply stable/datadog/templates/daemonset.yaml:3]: @gtaylor Sent a PR to your repo adding this env variable.

File: stable/datadog/README.md
Comment (irabinovitch) [reply stable/datadog/README.md:36]: @gtaylor  sent over a PR that documents the options.

**Comment by gtaylor** [comment]
@lachie83 @irabinovitch Sorry for the delay. I've merged the PRs that address the issues raised with this chart (thanks, @irabinovitch !). I think we're ready for another lookover.

I can squash this before we merge, or a committer can feel free to do the same. Whichever works better for you.

**Comment by gtaylor** [comment]
@irabinovitch Whoops, looks like you need to sign the CNCF CLA as shown above.

File: stable/datadog/templates/daemonset.yaml
Comment (gtaylor) [reply stable/datadog/templates/daemonset.yaml:3]: Merged ira's PR!

File: stable/datadog/README.md
Comment (gtaylor) [reply stable/datadog/README.md:36]: Merged!

**Comment by lachie83** [comment]
Perfect. I'll review again

**Comment by viglesiasce** [comment]
I just tested this and was able to get both docker and K8s events without any issues.

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

**Comment by irabinovitch** [comment]
Sorry folks. I thought I was helping, but probably slowed things down given CLA.  I hope to have that signed ASAP.

**Comment by lachie83** [comment]
@irabinovitch Are you happy with the state of this chart?

**Review by lachie83 — APPROVED** [review]

**Comment by irabinovitch** [comment]
@lachie83 This is great.  The only remaining change id make is the the dependency on kube-state-metrics.

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

**Comment by lachie83** [comment]
@irabinovitch Any update on getting the CLA signed. Should we bump the kube-state-metrics to another PR?

**Comment by viglesiasce** [comment]
@gtaylor we can squash and merge once the CLA is signed by @irabinovitch

**Comment by irabinovitch** [comment]
Our CLA should be all set now.

**Comment by viglesiasce** [comment]
I think we should merge this as is and take any updates in another PR. All in favor?

**Comment by irabinovitch** [comment]
Works for me.

**Pull request merged**
